### ISSUE TITLE
Define CMAbstractModel instances as a singleton?

### ISSUE THREAD

**Issue by SergeStinckwich** [cormas/cormas#160]
We should define ``CMAbstractModel`` (previously CormasModel) instances as real singleton, because information like timeStep are store on class side to share the the information with agents. So we have duplication of methods/attributes at instance level and class level.

Solution: use singleton pattern for ``CMAbstractModel`` so for example timeStep is ask to the instance and not the class.

Make sense?

After that there is another problem, about the fact that CORMAS could not run the same model multiple times, because we are using Singleton: #16

**Comment by hernanmd** [comment]
So that means no two or N CormasModels could be opened at the same time, so models will not be comparable during live development/simulation in the same image? 

I don't know if that would make sense for modelers. Maybe @pierbo can enlighten us :)

**Comment by SergeStinckwich** [comment]
This is already the case, this is not possible to open/run multiple CormasModel, because a lot of information is store on the class side (like timeStep for agents). This will I propose to have a clean singleton implementation of CormasModel and then remove this limitation.

**Comment by hernanmd** [comment]
I understand, ok then looks fine for me!
Thanks

**Comment by tomooda** [comment]
created a PR (https://github.com/cormas/cormas/pull/183). We can run multiple Conway models of the same class. Yey!
![スクリーンショット 2020-07-15 20 07 54](https://user-images.githubusercontent.com/836308/87538334-02796580-c6d7-11ea-96cb-701ce1e41f40.png)

**Comment by SergeStinckwich** [comment]
Great work Tomo ! But not sure too understand how it works :-)
Because this issue was about having ``CMAbstractModel`` as singleton.

**Comment by tomooda** [comment]
I don't think we need singleton, but there might exist old models that use classes to store global (model-wise) variables. Maybe It's nice to define `isSingleton` method that such old models can override.

**Comment by olekscode** [comment]
`CMAbstractModel` was separated in two classes: `CMAbstractModel` and `CMSimulation`
The new `CMSimulation` class has only instance variables and leaves no side effects between the simulation runs

In other words, when you run a simulation, this creates a new instance of `CMSimulation` and that instance has an instance variable `currentTimeStep`
